src/clinical/clinical-service.ts

```diff
@@ -219,27 +225,46 @@ export const getPaginatedClinicalData = async (
   query: PaginatedClinicalQuery,
 ) => {
   if (!programId) throw new Error('Missing programId!');
-  const start = new Date().getTime() / 1000;
 
   const allSchemasWithFields = await dictionaryManager.instance().getSchemasWithFields();
   // Get all donors + records for given entity
   const { donors, totalDonors } = await donorDao.findByPaginatedProgramId(programId, query);
 
   const taskToRun = WorkerTasks.ExtractEntityDataFromDonors;
-  const taskArgs = [donors as Donor[], totalDonors, allSchemasWithFields, query];
+  const taskArgs = [donors as Donor[], totalDonors, allSchemasWithFields, query.entityTypes, query];
```

Comment:
the `ExtractEntityDataFromDonors` task was updated torequire the `PaginatedQuery` in place of the `ClinicalDonorEntityQuery` since the new request by DonorId doesn't include much of that request data. The only additional information used from that query type was the list of entities to include in the request, which is now passed as a separate argument `query.entityTypes`.

Reply:
I see the type for query implies the property (`entityTypes`) is guaranteed to be present.
is it safe to assume its actual presence is validated on runtime elsewhere?